pennylane/gradients/parameter_shift.py

```diff
@@ -327,12 +367,15 @@ def processing_fn(results):
             res = results[start : start + num_tapes] if batch_size is None else results[start]
             start = start + num_tapes
 
-            g = _evaluate_gradient(res, data, broadcast, r0, scalar_qfunc_output)
+            g = _evaluate_gradient(tape, res, data, broadcast, r0, scalar_qfunc_output)
 
             grads.append(g)
-            # This clause will be hit at least once (because otherwise all gradients would have
-            # been zero), providing a representative for a zero gradient to emulate its type/shape.
-            zero_rep = qml.math.zeros_like(g)
+            if not qml.active_return() or not multi_measure:
+                # This clause will be hit at least once (because otherwise all gradients would have
+                # been zero), providing a representative for a zero gradient to emulate its type/shape.
+                zero_rep = qml.math.zeros_like(g)
+            else:
+                zero_rep = tuple(qml.math.zeros_like(grad_component) for grad_component in g)

```

Comment:
Can we take this out of the `for data in gradient_data` loop? It seems like `zero_rep` is being overwritten every iteration without being used.

Reply:
Good point! I've updated the logic to update `zero_rep` once only by setting it to `None` before the loop and checking `if zero_rep is None` before any updates.

Reply:
if the zero rep should just be based on the first element of this loop, I'd set it after the loop. ie. the if-else block replacing `g` with `grads[0]`

Reply:
Great suggestion, applied it. Thanks :)